src/DetailsView/components/save-assessment-factory.tsx

```diff
@@ -0,0 +1,22 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import * as React from 'react';
+import { CommandBarProps } from 'DetailsView/components/details-view-command-bar';
+import { DetailsViewActionMessageCreator } from '../actions/details-view-action-message-creator';
+import { SaveAssessmentButton } from 'DetailsView/components/save-assessment-button';
+
+export type SaveAssessmentFactoryDeps = {
+    detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
+};
+
+export type SaveAssessmentFactoryProps = CommandBarProps & {
+    deps: SaveAssessmentFactoryDeps;
+};
+
+export function getReportForAssessment(props: SaveAssessmentFactoryProps): JSX.Element {
```

Comment:
You might want to consider a different name for this and the below function--maybe something more related to what it's returning, like "getSaveButtonFor<Assessment or Fastpass>"?

Reply:
Updated, thank you!